src/core/meshcop/commissioner.hpp

```diff
@@ -74,6 +75,160 @@ class Commissioner : public InstanceLocator, private NonCopyable
         kStateActive   = OT_COMMISSIONER_STATE_ACTIVE,   ///< Active Commissioner.
     };
 
+    /**
+     * This enumeration type represents Joiner Event.
+     *
+     */
+    enum JoinerEvent : uint8_t
+    {
+        kJoinerEventStart     = OT_COMMISSIONER_JOINER_START,
+        kJoinerEventConnected = OT_COMMISSIONER_JOINER_CONNECTED,
+        kJoinerEventFinalize  = OT_COMMISSIONER_JOINER_FINALIZE,
+        kJoinerEventEnd       = OT_COMMISSIONER_JOINER_END,
+        kJoinerEventRemoved   = OT_COMMISSIONER_JOINER_REMOVED,
+    };
+
+    /**
+     * This enumeration represents the resign mode and is used as input to `Stop()` method.
+     *
+     */
+    enum ResignMode : uint8_t
+    {
+        kDoNotResign, ///< Do not resign.
+        kResign,      ///< Resign.
```

Comment:
```suggestion
        kPassiveResign, ///< Do not resign.
        kActiveResign,      ///< Resign.
```
Not sure if these names are proper, either. 

We were using `/* aResign */ false` in cases where the commissioner session does not exist on Leader, so that the Commissioner does not need to Resign actively.

Reply:
Good point. I change them to `kSendKeepAliveToResign` or `DoNotSendKeepAlive` which describes what we do.

Reply:
SGTM